src/NumericalAlgorithms/Interpolation/CubicSpline.hpp

```diff
@@ -0,0 +1,60 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <gsl/gsl_spline.h>
+#include <vector>
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace intrp {
+/*!
+ * \ingroup NumericalAlgorithmsGroup
+ * \brief A cubic spline interpolation class
+ *
+ * The class builds a cubic spline interpolant using the `x_values` and
+ * `y_values` passed into the constructor. For details on the algorithm see
+ * the GSL documentation on `gsl_interp_cspline`.
+ *
+ * Here is an example how to use this class:
+ *
+ * \snippet Test_CubicSpline interpolate_example
+ */
+class CubicSpline {
```

Comment:
Is the plan for 2d splines to name them `BicubicSpline`? I'm asking to avoid having to rename this class in the future. we will almost certainly need 2d and 3d splines at some point.

Reply:
`BicubicSpline` would fit GSL's naming https://www.gnu.org/software/gsl/doc/html/interp.html#c.gsl_interp2d_bicubic. Probably better than templating this class on dimension

Reply:
Okay, sounds good! Just wanted to make sure it was briefly thought about and discussed. Thanks!